src/components/SigningModals/TransactionJourney.js

```diff
@@ -0,0 +1,265 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { css, keyframes } from 'styled-components'
+import { textStyle, useTheme, GU, RADIUS } from '@aragon/ui'
+
+const PREVIOUS = 'PREVIOUS'
+const CURRENT = 'CURRENT'
+const NEXT = 'NEXT'
+
+const ITEM_SPACING = 3.5 * GU
+
+const pulseAnimation = keyframes`
+  from {
+    transform: scale3d(0.25, 0.25, 1);
+    opacity: 0.5;
+  }
+
+  to {
+    transform: scale3d(1.25, 1.25, 1);
+    opacity: 0;
+  }
+`
+
+function getItemStatus(i, activeItem) {
+  if (i < activeItem) {
+    return PREVIOUS
+  }
+
+  if (i > activeItem) {
+    return NEXT
+  }
+
+  return CURRENT
+}
+
+function TransactionJourney({ items, activeItem, pulse, themeColor }) {
+  const theme = useTheme()
+  const color = themeColor || theme.accent
+
+  const renderItems = () =>
+    items.map(([title, { icon, content }], i) => {
+      const status = getItemStatus(i, activeItem)
+
+      return (
+        <JourneyItem
+          key={i}
+          title={title}
+          icon={icon}
+          status={status}
+          isLastItem={i === items.length - 1}
+          pulse={pulse}
+          themeColor={color}
+        >
+          {content}
+        </JourneyItem>
+      )
+    })
+
+  return (
+    <ul
+      css={`
+        list-style: none;
+      `}
+    >
+      {renderItems()}
+    </ul>
+  )
+}
+
+TransactionJourney.defaultProps = {
+  items: [],
+  activeItem: 0,
+  pulse: false,
+}
+
+TransactionJourney.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.array),
+  activeItem: PropTypes.number,
+  pulse: PropTypes.bool,
+  themeColor: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+}
+
+/* eslint-disable react/prop-types */
+function PulsePip({ className, size, themeColor }) {
+  return (
+    <div
+      className={className}
+      css={`
+        width: ${size}px;
+        height: ${size}px;
+
+        &:after,
+        &:before {
+          content: '';
+          position: absolute;
```

Comment:
This is being "hidden" by the fact that we're using `PulsePip` with a `position: absolute` declaration, but I think this child positioning may break if the `PulsePip` isn't at least relatively positioned?

In that case, we may want to layer another `div` to apply this, to be safe.

Reply:
Good catch!